neo4j/internal/bolt/hydrator.go

```diff
@@ -952,28 +973,87 @@ func parseProfile(profilex map[string]any) *db.ProfiledPlan {
 	return plan
 }
 
+func parseInputPosition(m map[string]any) *db.InputPosition {
+	if m == nil {
+		return nil
+	}
+	pos := &db.InputPosition{}
+	if i, ok := m["column"].(int64); ok {
+		pos.Column = int(i)
+	}
+	if i, ok := m["line"].(int64); ok {
+		pos.Line = int(i)
+	}
+	if i, ok := m["offset"].(int64); ok {
+		pos.Offset = int(i)
+	}
```

Comment:
```suggestion
	if i, ok := m["column"].(int64); ok {
		pos.Column = i
	}
	if i, ok := m["line"].(int64); ok {
		pos.Line = i
	}
	if i, ok := m["offset"].(int64); ok {
		pos.Offset = i
	}
```
Why is this a lossy cast? Shouldn't `InputPosition.Offset` etc. be int64 instead?

Reply:
Honestly no idea this was refactored away from the parseNotification function/original code we had, I assumed there was some hidden logic in why it was implemented way in the first place.

Reply:
One thing to think about is that the InputPosition interface exposes all ints, not int64, so this would be a breaking change for the older notification stuff.

Reply:
Packstream deals with int64 - this is an abstraction, other drivers do this.